### PR TITLE
Published Hidden Filter

### PR DESCRIPTION
**Pull request by visitharri**
Day, month and year have been separated in different inputs in the refactored code. Do you think this is okay, or should I consider drop down selection, where user must choose from only available options?

### CODE REVIEW

File: frontend/src/components/MapFiltersMenu/MapFiltersMenu.tsx [hunk 1]
```diff
@@ -85,8 +85,6 @@ const MapFiltersMenu = ({
     };
 
     const resetFilters = () => {
-        mapFilters.publicSince = "";
-        mapFilters.publicUntil = "";
         setMapFilters(initFilters);
         updateFilters({ ...initFilters });
     };
```
Comment (visitharri) [inline frontend/src/components/MapFiltersMenu/MapFiltersMenu.tsx]: Clicking "Tyhjennä" (Reset) does empty the filters, but text input content does not disappear. It does so with "name contains" filter text input, but here in "published hidden" filter it does not happen. Well, it happens, if you update the map by clicking "Etsi alueelta" (search in the visible area), but does not by using "Tyhjennä" (Reset).

**Comment by visitharri** [comment]
Here are still some unfixed issues: 
1) Text input content does not disappear when clicking "Tyhjennä" (Reset). Check "small fix".
2) Need to add some regex for text inputs.

File: frontend/src/components/MapFilters/PublishedHiddenFilter/PublishedHiddenFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                    </div>
+
+                    <div className="public-text-input">
+                        {"Mihin: "}
+                        <input
+                            name="publicUntil"
+                            type="text"
+                            id="publicUntil"
+                            onChange={modifyPublicUntil}
+                            value={publicUntil}
```
Comment (kemm0) [inline frontend/src/components/MapFilters/PublishedHiddenFilter/PublishedHiddenFilter.tsx:151]: The issue where the value is not updated correctly after reset can be solved by giving it an empty string if publicUntil is undefined. Like this:

value={publicUntil || ""}

**Review by kemm0 — CHANGES_REQUESTED** [review]

**Comment by visitharri** [comment]
Day, month and year have been separated in different inputs in the refactored code. Is this okay, or should I consider drop down selection, where user must choose from only available options?

File: frontend/src/components/MapFilters/PublishedHiddenFilter/PublishedHiddenFilter.tsx (hunk 2 above)
Comment (visitharri) [reply frontend/src/components/MapFilters/PublishedHiddenFilter/PublishedHiddenFilter.tsx]: Text input content now disappears when clicking "Tyhjennä" (Reset), but only when closing and opening filters menu again.

File: frontend/src/components/MapView/MapView.tsx [hunk 3]
```diff
@@ -23,6 +24,7 @@ const MapView = () => {
         const bounds = map.getBounds();
         const centerPoint = map.getCenter();
         let kmDistance: number;
+        if(!filters.isPublic) filters.isPublic = DEFAULT_IS_PUBLIC;
```
Comment (Mel-dev-hub) [inline frontend/src/components/MapView/MapView.tsx:27]: what is this part for?

Comment (visitharri) [reply frontend/src/components/MapView/MapView.tsx:27]: There are 3 radio buttons here in this filter. The first one is default. So if nothing changed, this default radio must still be sent

Comment (Mel-dev-hub) [reply frontend/src/components/MapView/MapView.tsx:27]: got it

**Review by Mel-dev-hub — APPROVED** [review]
looks good overall

File: frontend/src/components/MapFilters/PublishedHiddenFilter/PublishedHiddenFilter.tsx (hunk 2 above)
Comment (visitharri) [reply frontend/src/components/MapFilters/PublishedHiddenFilter/PublishedHiddenFilter.tsx]: This has been changed

**Review by kemm0 — APPROVED** [review]

**Pull request merged**